api/src/main/java/com/redhat/ipaas/api/IPaasEntity.java

```diff
@@ -15,9 +15,15 @@
  */
 package com.redhat.ipaas.api;
 
-public interface IPaasEntity {
+public abstract class IPaasEntity {
 
-	public String getId();
-	public void setId(String id);
+	abstract public String getId();
+	abstract public void setId(String id);
+	
+	@Override
+	public boolean equals(Object obj) {
```

Comment:
Because they're not equal just because of ID - you're just checking that you don't have items with duplicate IDs in the list. That is not object equality. This will hurt us if we ever want to use other data structures, e.g. Sets, that contain these objects.

Plus what is more readable about `list.contains(obj)` than `containsId(list, obj)` where `containsId` is a utility function to do the check. To my mind that's arguably _more_ readable.